server/src/main/java/io/crate/planner/operators/EquiJoinDetector.java

```diff
@@ -76,54 +76,46 @@ private static class Visitor extends SymbolVisitor<Context, Void> {
         public Void visitFunction(Function function, Context context) {
             String functionName = function.name();
             switch (functionName) {
-                case AndOperator.NAME:
-                    for (Symbol arg : function.arguments()) {
-                        arg.accept(this, context);
+                case NotPredicate.NAME -> {
+                }
+                case OrOperator.NAME -> {
+                    context.isHashJoinPossible = false;
+                }
```

Comment:
```suggestion
                case NotPredicate.NAME, OrOperator.NAME -> {
                    return null;
                }
```

Reply:
See my comment here: https://github.com/crate/crate/pull/15687#discussion_r1530254255

Reply:
example:
```
@Test
    public void testOr() {
        Symbol joinCondition = sqlExpressions.asSymbol("(t1.a = t2.b or t1.x = t2.y) and t1.i = t2.i");
        assertThat(EquiJoinDetector.isHashJoinPossible(JoinType.INNER, joinCondition), is(false));
    }
```
this  fails if we exit and don't reset the flag to false.

Reply:
This one also fails on the previous version before this pr.

Reply:
I think we need more testing.